plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1624,12 +1613,21 @@ private static void appendAddFileEntries(TransactionLogWriter transactionLogWrit
                             info.getSize(),
                             info.getCreationTime(),
                             dataChange,
-                            Optional.of(serializeStatsAsJson(info.getStatistics())),
+                            Optional.of(serializeStatsAsJson(statisticsWithExactNames)),
                             Optional.empty(),
                             ImmutableMap.of()));
         }
     }
 
+    private static Optional<Map<String, Object>> toOriginalColumnNames(Optional<Map<String, Object>> statistics, Map<String, String> exactColumnNames)
```

Comment:
exactColumnNames -> lowerCaseToExactColumnNames (or toOriginalColumnNames)
`Map<String, String>` -> `Map</* lowercase*/ String, String>` (or switch to `CanonicalColumnName` as map keys)